File: data/src/main/java/com/example/data/mapper/RoomMapper.java
package com.example.data.mapper;

import com.example.data.model.APODRoom;
import com.example.domain.model.APODEntity;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.inject.Inject;

/**
 * Реализация маппинга из room в entity и наоборот
 */

public class RoomMapper implements IMapper<APODEntity, APODRoom> {

    @Inject
    public RoomMapper() {
    }

    @NotNull
    @Override
    public APODRoom mapFromEntity(@NotNull APODEntity type) {
        return new APODRoom(
                type.getDate(),
                type.getExplanation(),
                type.getTitle(),
                type.getUrl(),
                type.getCopyright());
    }

    @NotNull
    @Override
    public APODEntity mapToEntity(@Nullable APODRoom type) {
        APODEntity apodEntity;
        if (type != null) {
            apodEntity = new APODEntity(
                    type.getDate(),
                    type.getExplanation(),
                    type.getTitle(),
                    type.getUrl(),
                    type.getCopyright());
        } else {
            apodEntity = new APODEntity(
                    "",
                    "",
                    "",
                    "",
                    "");
        }
        return apodEntity;
    }
}

File: data/src/main/java/com/example/data/mapper/RoomMapper.kt
package com.example.data.mapper

import com.example.data.model.APODRoom
import com.example.domain.model.APODEntity
import javax.inject.Inject

/**
 * Реализация маппинга из room в entity и наоборот
 */
open class RoomMapper @Inject constructor() : IMapper<APODEntity, APODRoom> {
    override fun mapFromEntity(type: APODEntity): APODRoom {
        return APODRoom(
                type.date,
                type.explanation,
                type.title,
                type.url,
                type.copyright)
    }

    override fun mapToEntity(type: APODRoom): APODEntity {
        return APODEntity(
                type.date,
                type.explanation,
                type.title,
                type.url,
                type.copyright ?: "")

    }
}